### PR TITLE
scheduler: remove old scheduler name

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/8379

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/schedulers/diagnostic_recorder.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ import (
 	sc "github.com/tikv/pd/pkg/schedule/config"
 	"github.com/tikv/pd/pkg/schedule/operator"
 	"github.com/tikv/pd/pkg/schedule/plan"
+	types "github.com/tikv/pd/pkg/schedule/type"
```
Comment (rleungx) [inline pkg/schedule/schedulers/diagnostic_recorder.go:26]: How about renaming it to types?

File: pkg/schedule/schedulers/diagnostic_recorder.go [hunk 2]
```diff
@@ -131,11 +132,11 @@ func (d *DiagnosticRecorder) GetLastResult() *DiagnosticResult {
 			}
 		} else if firstStatus == Pending {
 			// This is used to handle pending status because of reach limit in `IsScheduleAllowed`
-			resStr = fmt.Sprintf("%s reach limit", d.schedulerName)
+			resStr = fmt.Sprintf("%v reach limit", d.schedulerType)
 		}
 	}
 	return &DiagnosticResult{
-		Name:      d.schedulerName,
+		Name:      string(d.schedulerType),
```
Comment (rleungx) [inline pkg/schedule/schedulers/diagnostic_recorder.go:139]: What is the difference between `string(d.schedulerType)` and `d.schedulerType.String()`?

Comment (okJiang) [reply pkg/schedule/schedulers/diagnostic_recorder.go:139]: no difference

File: pkg/schedule/schedulers/diagnostic_recorder.go (hunk 1 above)
Comment (okJiang) [reply pkg/schedule/schedulers/diagnostic_recorder.go:26]: sgtm

File: pkg/schedule/schedulers/diagnostic_recorder.go (hunk 2 above)
Comment (rleungx) [reply pkg/schedule/schedulers/diagnostic_recorder.go:139]: Then shall we keep consistent with other places?

Comment (okJiang) [reply pkg/schedule/schedulers/diagnostic_recorder.go]: updated

**Review by rleungx — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Review by niubell — APPROVED** [review]

**Pull request merged**